src/olympia/versions/views.py

```diff
@@ -31,9 +31,9 @@
 @mobile_template('versions/{mobile/}version_list.html')
 @non_atomic_requests
 def version_list(request, addon, template, beta=False):
-    status_list = (amo.STATUS_BETA,) if beta else amo.VALID_FILE_STATUSES
+    status = amo.STATUS_BETA if beta else amo.STATUS_PUBLIC
     qs = (addon.versions.filter(channel=amo.RELEASE_CHANNEL_LISTED)
-          .filter(files__status__in=status_list)
+          .filter(files__status=status)
```

Comment:
If there is no public version, we should show the version pending review, as per https://github.com/mozilla/addons-server/issues/3704#issuecomment-252588543

Otherwise, it might look weird, that the listing page has a link to the add-on file but the versions page does not.